server-phoenix/lib/helios_web/controllers/events_api/events_api_controller.ex

```diff
@@ -34,6 +35,24 @@ defmodule HeliosWeb.EventsAPI.EventsAPIController do
     send_resp(conn, 201, "Created")
   end
 
+  defp post_pr(conn, params) do
+    unless(
+      Event
+      |> Event.pull_requests()
+      |> Event.with_external_id(params["event_id"])
+      |> Repo.exists?()
+    ) do
+      Repo.insert!(%Event{
+        source: :github_pull,
+        external_id: params["event_id"],
+        source_author: params["author"]
+      })
+      |> publish
+    end
```

Comment:
This is repeated from commit. Could we extract a method that passes in the different source?

Reply:
Was about to say that would be a rather verbose method as `:commit` and `:pr` need their own get_all query, and that it wouldn't work out the box for other `Event` types; but TIL you can pass functions in Elixir! Updated as follows:
```elixir
defp post_event(conn, get_all_of_source_type, source, external_id, source_author) do
    unless(
      Event
      |> get_all_of_source_type.()
      |> Event.with_external_id(external_id)
      |> Repo.exists?()
    ) do
      Repo.insert!(%Event{
        source: source,
        external_id: external_id,
        source_author: source_author
      })
      |> publish
...
```